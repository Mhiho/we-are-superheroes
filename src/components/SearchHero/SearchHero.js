import React from 'react';


const searchHero = props => {
    return (
      <div>
        <form  onSubmit={props.submit}>
          <input
            placeholder="type your superhero"
            onChange={props.change}
            type="search"
            value={props.term}
            />
            <input
            className="btn btn-md btn-primary"
            type="submit"
            value="search"
            />
          </form>
      </div>
    )
}

export default searchHero;