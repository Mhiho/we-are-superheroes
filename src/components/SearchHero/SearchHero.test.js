import React from 'react';
import renderer from 'react-test-renderer';

import SearchHero from './SearchHero';

it('renders correctly when term is empty string', () => {
  const tree = renderer.create(<SearchHero />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly where there is object as props', ()=>{
    const term = 'term'
    const tree = renderer.create(<SearchHero term={term} />).toJSON();
    expect(tree).toMatchSnapshot();
})
