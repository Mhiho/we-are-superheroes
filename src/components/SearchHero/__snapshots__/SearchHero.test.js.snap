// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when term is empty string 1`] = `
<div>
  <form>
    <input
      placeholder="type your superhero"
      type="search"
    />
    <input
      className="btn btn-md btn-primary"
      type="submit"
      value="search"
    />
  </form>
</div>
`;

exports[`renders correctly where there is object as props 1`] = `
<div>
  <form>
    <input
      placeholder="type your superhero"
      type="search"
      value="term"
    />
    <input
      className="btn btn-md btn-primary"
      type="submit"
      value="search"
    />
  </form>
</div>
`;
